pylint_plugins/conf_plugin.py

```diff
@@ -0,0 +1,90 @@
+
+import astroid
+
+from astroid import MANAGER
+from pylint import lint
+
+def __init__(self):
+  pass
+
+
+#didn't change/work on this yet
+def wrap_add_message(func):
+  def add_message(msg_descr, line=None, node=None, args=None, confidence=None):
+    kwargs = {'confidence': confidence} if confidence else {}
+    if (msg_descr != 'no-member' or
+        args[0:2] != ('Module', 'openhtf.conf')):
+      return func(msg_descr, line, node, args, **kwargs)
+    # We're sure it's a no-member warning for the openhtf.conf module.
+    try:
+      if args[2] in node.root()._declared_conf_keys:
+        # We found a declaration for the symbol, just skip this message.
+        return
+    except AttributeError:
+      pass
+
+    # TODO(maddychan): Figure out how to add a custom failure message here.
+    return func('no-member', line, node, args, **kwargs)
+
+  return add_message
+
+#storing a reference to the openhtf.conf module node
+clss = []
+clss_locals = {}
+current_root = []
+
+def trans_expr(node):
+  if (isinstance(node.func, astroid.Attribute) and
+      isinstance(node.func.expr, astroid.Name) and
+      node.func.expr.name == 'conf' and
+      node.func.attrname == 'Declare'):
+
+    #import pdb
+    #pdb.set_trace()
+
+    #keeping track of the current root, refreshing the locals if it changes
+    if current_root == []:
+    	current_root.append(node.root())
+    elif current_root[0] != node.root():
+    	current_root[0] = node.root()
+    	clss[0].locals = clss_locals
+
+    #adding the conf attribute to the module's locals
+    if node.args and clss:
+    	clss[0].locals[node.args[0].value] = [None]
```

Comment:
Does this have to be `[None]` or can it just be `None`?  you just need `node.args[0].value` to be in the `locals`, right?  Also, as a side note, it doesn't *have* to be `args[0]`, it can technically be passed by keyword, hence the annoying logic below.  We don't currently actually do that anywhere, but it would be weird if pylint exploded over it.